lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/BuildJsonHttpDeserializer.kt

```diff
@@ -88,18 +92,18 @@ internal fun <T> buildJsonBatchHttpDeserializer(
         }
 
         val jsonResponses = Json.parseToJsonElement(response.body).jsonArray
-        val responses = jsonResponses.map {
-            val deserializationStrategy = deserializationStrategies[it.jsonObject["id"]!!.jsonPrimitive.int]
-            deserializationJson.decodeFromJsonElement(
+        val orderedResults = MutableList<T?>(jsonResponses.size) { null }
+
+        jsonResponses.forEach { jsonElement ->
+            val id = jsonElement.jsonObject["id"]!!.jsonPrimitive.int
+            val deserializationStrategy = deserializationStrategies[id]
+            val jsonRpcResponse = deserializationJson.decodeFromJsonElement(
                 JsonRpcResponse.serializer(deserializationStrategy),
-                it,
+                jsonElement,
             )
+            orderedResults[id] = extractResult(jsonRpcResponse, response.body, jsonElement.toString())
         }
 
-        val results = responses.sortedBy { it.id }.zip(jsonResponses)
-            .map { (jsonRpcResponse, jsonResponse) ->
-                extractResult(jsonRpcResponse, response.body, jsonResponse.toString())
-            }
-        results
+        orderedResults.filterNotNull()
```

Comment:
I get the idea. But I'm not sure the linear ordering here is worth introducing mutability and breaking null-safety, as these batch requests are very unlikely to have considerably big number of requests. 

Are you sure there's no solution for this to maintain these two?

Reply:
Discussed offline, decided to leave the current approach 👍🏻